packages/material-ui/src/Accordion/Accordion.js

```diff
@@ -1,64 +1,98 @@
 import * as React from 'react';
 import { isFragment } from 'react-is';
 import PropTypes from 'prop-types';
-import clsx from 'clsx';
-import { chainPropTypes } from '@material-ui/utils';
+import { deepmerge, chainPropTypes } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
 import Collapse from '../Collapse';
-import Paper from '../Paper';
-import withStyles from '../styles/withStyles';
 import AccordionContext from './AccordionContext';
 import useControlled from '../utils/useControlled';
+import accordionClasses, { getAccordionUtilityClass } from './accordionClasses';
 
-export const styles = (theme) => {
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...(!styleProps.square && styles.rounded),
+    ...(styleProps.expanded && styles.expanded),
+    ...(styleProps.disabled && styles.disabled),
+    [`& .${accordionClasses.region}`]: styles.region,
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+    rounded: ['rounded'],
+    expanded: ['expanded'],
+    disabled: ['disabled'],
+    region: ['region'],
+  };
+
+  return composeClasses(slots, getAccordionUtilityClass, classes);
+};
+
+const AccordionRoot = experimentalStyled(
+  'Paper',
+  {},
+  {
+    name: 'MuiAccordion',
+    slot: 'Root',
+    overridesResolver,
+  },
+)(({ theme, styleProps }) => {
   const transition = {
     duration: theme.transitions.duration.shortest,
   };
 
   return {
     /* Styles applied to the root element. */
-    root: {
-      position: 'relative',
-      transition: theme.transitions.create(['margin'], transition),
-      overflowAnchor: 'none', // Keep the same scrolling position
+    position: 'relative',
+    transition: theme.transitions.create(['margin'], transition),
+    overflowAnchor: 'none', // Keep the same scrolling position
+    '&:before': {
+      position: 'absolute',
+      left: 0,
+      top: -1,
+      right: 0,
+      height: 1,
+      content: '""',
+      opacity: 1,
+      backgroundColor: theme.palette.divider,
+      transition: theme.transitions.create(['opacity', 'background-color'], transition),
+    },
+    '&:first-child': {
       '&:before': {
-        position: 'absolute',
-        left: 0,
-        top: -1,
-        right: 0,
-        height: 1,
-        content: '""',
-        opacity: 1,
-        backgroundColor: theme.palette.divider,
-        transition: theme.transitions.create(['opacity', 'background-color'], transition),
+        display: 'none',
       },
+    },
+    /* Styles applied to the root element if `expanded={true}`. */
+    ...(styleProps.expanded && {
+      margin: '16px 0',
       '&:first-child': {
```

Comment:
Just pushed a commit that I *think* applies the changes requested, does that look accurate?

Reply:
Yep, looks like https://github.com/mui-org/material-ui/pull/24518/commits/ce9c0288e3099a79da5c478a71fa55d684cff86c fixes this :)

Reply:
Although, now that I see the changes, after we converted the `styleProp.expanded` to class selector it may not have been necessary. Anyway, let's leave it as is

Reply:
Hmm, it seems that the styles weren't properly applied with this commit, so the top accordion panel was getting a top margin.  Moving it back seemed to resolve it, maybe something to do with the nesting order of the properties?